### PR TITLE
[Merged by Bors] - Snappy additional sanity checks

### PR DESCRIPTION
**Pull request by pawanjay176**
## Issue Addressed

N/A

## Proposed Changes

Adds the following check from the spec

> A reader SHOULD NOT read more than max_encoded_len(n) bytes after reading the SSZ length-prefix n from the header.

### CODE REVIEW

File: beacon_node/eth2_libp2p/src/rpc/codec/ssz_snappy.rs [hunk 1]
```diff
@@ -409,11 +412,50 @@ impl<TSpec: EthSpec> OutboundCodec<RPCRequest<TSpec>> for SSZSnappyOutboundCodec
...
+        match reader.read(buf) {
+            Ok(0) => break,
+            Ok(n) => {
+                let tmp = buf;
+                buf = &mut tmp[n..];
+            }
+            Err(ref e) if e.kind() == ErrorKind::Interrupted => {}
+            Err(e) => return Err(e),
+        }
+        count += reader.get_ref().position() - i;
```
Comment (AgeManning) [inline beacon_node/eth2_libp2p/src/rpc/codec/ssz_snappy.rs:444]: can the position here ever be less than the previous position and underflow?

**Review by AgeManning — COMMENTED** [review]
I'm a little confused about the wrapper. 

You can calculate the `max_enccoded_length`. Can we not just check that `src.len()` < than this value?

File: beacon_node/eth2_libp2p/src/rpc/codec/ssz_snappy.rs (hunk 1 above)
Comment (pawanjay176) [reply beacon_node/eth2_libp2p/src/rpc/codec/ssz_snappy.rs]: Checked the snappy read implementation and it always advances the reader forward as far as I can tell. However, added an underflow check to be safe.

**Review by AgeManning — APPROVED** [review]
Looks good to me

**Pull request closed without merging**